designs/containers/workload-discovery/workload-discovery.md

```diff
@@ -0,0 +1,185 @@
+# OSC Container Support - Workload Discovery
+This document describes the design changes needed for OSC to support microsegmentation of workloads on a Kubernetes(K8s) cluster.  Microsegmentation is implemented on OSC through the creation of [security groups](#osc-security-groups) which identify the workloads to be protected by a VNF as defined by the security policies applied to the group.  
+Applying the security policies to the group by deploying the VNF and configuring the traffic redirection (aka, steering) is out of scope for this document and will be covered on complimentary design documents. 
+
+## Assignees
+Emanoel Xavier - https://github.com/emanoelxavier
+
+## Background
+OSC currently supports protection of workloads hosted on VM based virtualization environments. This work expands the OSC scope to include also container environments orchestrated by [Kubernetes](#kubernetes-home). The adoption of container technologies is becoming more and more widespread due to its benefits such as easy maintainability, reusability, and minimal overhead with K8s being one of the most popular container orchestrators.  
+For the first release of the OSC containers support the focus will be mostly on the integration points between OSC, the virtualization environment and the software defined network(SDN) controller services needed to discovering the workloads to be protected, deploying the security VNFs and performing traffic redirection through a demonstrable E2E flow.  
+
+## Constraints and Assumptions
+### Kubernetes Version and Configuration
+* **Kubernetes Version:** 1.6.7. 
+* **Network Layer**:  [OVN](#ovn-kubernetes) with [SFC](#ovs-sfc) enabled or Nuage.
+* **Authentication**: For this first release the access to the Kubernetes API is allowed over `http` and **without user authentication**.  This should be acceptable as the intent of this release is for prototype/demonstration only rather than production readiness. For the longer run we want to allow only `https` access with authenticated users, this should be possible with [Fabric8](#fabric8-credential-example), the java SDK being adopted. 

```

Comment:
The demonstration of this e2e flow will be possible in third party environments, however it should not be a production but a prototype or staging environment instead. Security wise requiring user credentials does not harden the communication much since it is over http anyways. Also, if we use OVN the experimental setup steps for the k8s cluster does not enable authentication, so this aligns as well.   
Since this does not impact the functional e2e I would consider lower priority at the moment. In any case, I will track the auth work with a separate spike and user stories and if we can fit we can update this doc and address it within this release.